### PR TITLE
Side Nav Unit Tests

### PR DESCRIPTION
**Pull request by marshmalien**
##### SUMMARY
Added unit tests to cover the following directives:
* Layout
* Side Nav
* Side Nav Item

### CODE REVIEW

File: awx/ui/client/test/unit/side-nav-item.spec.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        });
+
+        it('go() should call $state.go()', angular.mock.inject((_$state_) => {
+            spyOn(_$state_, 'go');
+            SideNavItemCtrl.go();
+            expect(_$state_.go).toHaveBeenCalled();
+            expect(_$state_.go).toHaveBeenCalledWith('dashboard', jasmine.any(Object), jasmine.any(Object));
+        }));
+
+        it('should load name, icon, and route from scope', () => {
```
Comment (jlmitch5) [inline awx/ui/client/test/unit/side-nav-item.spec.js:55]: this makes sense to me!

File: awx/ui/client/test/unit/side-nav-item.spec.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            scope.$digest();
+            expect(SideNavItemCtrl.isRoute).toBe(true);
+
+            current = {'name': 'inventories'};
+            $rootScope.$broadcast('$stateChangeSuccess', current);
+            scope.$digest();
+            expect(SideNavItemCtrl.isRoute).toBe(false);
+        });
+
+        it('go() should call $state.go()', angular.mock.inject((_$state_) => {
```
Comment (jlmitch5) [inline awx/ui/client/test/unit/side-nav-item.spec.js:48]: I hadn't seen the use of angular.mock.inject within an `it` test function's callback, but I like that it is explicitly allowing you to spy on the thing you care about for the particular test.  Neat stuff!

File: awx/ui/client/test/unit/layout.spec.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                scope.$digest();
+                expect(controller.licenseIsMissing).toBe(true);
+
+                $rootScope.licenseMissing = false;
+                scope.$digest();
+                expect(controller.licenseIsMissing).toBe(false);
+            });
+        });
+
+        describe('getString()', () => {
```
Comment (jlmitch5) [inline awx/ui/client/test/unit/layout.spec.js:128]: great testing of all the functionality within the getString function.

**Review by jlmitch5 — APPROVED** [review]
Fantastic job on these unit tests!

**Comment by gconsidine** [comment]
Looks very good, @marshmalien. I have some feedback but I'm going to give the tests a few runs and verify a few things. Brbz.

File: awx/ui/client/test/unit/layout.spec.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            it('calls ComponentsStrings get() method', angular.mock.inject((_ComponentsStrings_) => {
+                spyOn(_ComponentsStrings_, 'get');
+                controller.getString('VIEW_DOCS')
+                expect(_ComponentsStrings_.get).toHaveBeenCalled();
+            }));
+
+            it('ComponentsStrings get() method  should throw an error', () => {
```
Comment (jakemcdermott) [inline awx/ui/client/test/unit/layout.spec.js]: The name of this test case needs some more context: "It should throw an error when no matching key is given"

File: awx/ui/client/test/unit/side-nav.spec.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        it('isExpanded defaults to false', () => {
+            expect(sideNavCtrl.isExpanded).toBe(false);
+        });
+
+        it('toggleExpansion()', () => {
+            sideNavCtrl.toggleExpansion();
+            expect(sideNavCtrl.isExpanded).toBe(true);
+
+            sideNavCtrl.toggleExpansion();
+            expect(sideNavCtrl.isExpanded).toBe(false);
```
Comment (jakemcdermott) [inline awx/ui/client/test/unit/side-nav.spec.js:56]: when testing anything that toggles, do it at least 4 times:

off **-1->** on **-2->** off **-3->** on **-4->** off

File: awx/ui/client/test/unit/side-nav-item.spec.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        angular.mock.module('at.lib.services')
+        angular.mock.module('at.lib.components')
+    });
+
+    beforeEach(angular.mock.inject((_$compile_, _$rootScope_) => {
+        $compile = _$compile_;
+        $rootScope = _$rootScope_;
+        scope = $rootScope.$new();
+
+        element = angular.element('<at-layout><at-side-nav><at-side-nav-item icon-class="fa-tachometer" route="dashboard" name="DASHBOARD"></at-side-nav-item></at-layout></at-side-nav><');
```
Comment (jakemcdermott) [inline awx/ui/client/test/unit/side-nav-item.spec.js]: the last `<` in `...<at-side-nav><` here might be a typo

Comment (marshmalien) [reply awx/ui/client/test/unit/side-nav-item.spec.js]: Ah. Thanks for catching that.

**Review by jakemcdermott — APPROVED** [review]

**Pull request merged**